cvat/apps/engine/views.py

```diff
@@ -1746,7 +1746,13 @@ def perform_destroy(self, instance):
         if instance.type != JobType.GROUND_TRUTH:
             raise ValidationError("Only ground truth jobs can be removed")
 
-        return super().perform_destroy(instance)
+        segment = instance.segment
+        if segment:
```

Comment:
In which case segment can be None?

Reply:
I do not have any ideas here, but moved this logic as is from `Job.delete` method 
Maybe @zhiltsov-max may comment

Reply:
I don't think this should happen in practice, it looks more like a default safe removal pattern.

Reply:
In that case, If this should not happen in practice, does such handling look like silent error suppression? (for me - yes). But, @zhiltsov-max, why do we need to call segment.delete() directly here? It will be deleted anyway after job deletion.

Reply:
In our implementation, segments own jobs, so they need to be deleted explicitly. I agree that it's currently meaningless to check that the segment is none, as it's prohibited by the field definition.

Reply:
Returned this code to model, I believe that would be a little bit safer if we remove job from admin panel for example